R/MEND2_MILLEN.R

```diff
@@ -0,0 +1,139 @@
+#' Define the 2 P pool MEND fluxes that uses the ... TODO figure out what millen stands for
+#'
+#' \code{MEND2_MILLEN_fluxes} Defines a system of equations that
+#' describe the state of the fluxes between the pools from the 2 pool version of
+#' \href{https://doi.org/10.1890/12-0681.1}{Wang et al. 2013}. By retruning a list of named functions.
+#'
+#' @param state A numeric vector of the different MEND carbon pool states.
+#' @param parms A data frame of the parameters.
+#' @return A list of functions that calculate the fluxes between the MEND 2 carbon pools that uses
+#' TODO doc uptake.
+#' @family MEND2_MILLEN
+#' @family carbon flux functions
+#' @importFrom assertthat assert_that has_name
+#' @noRd
+MEND2_MILLEN_fluxes  <- function(state, parms){
+
+  # Check inputs
+  assert_that(has_name(x = state, which = c("P1", "P2", "M", "B", "D1", "D2", "Q1",
+                                            "Q2", "EP1", "EP2", "EM", "IC", "Tot")))
+  assert_that(has_name(x = parms, which = c("parameter", "description", "units", "value")))
+  assert_that(has_name(x = parms, which = c("parameter",  "value")))
+
+  req <- c('E.c1', 'E.c2', 'V.d1', 'V.d2', 'K.d1', 'K.d2', 'V.p1', 'V.p2', 'K.p1', 'K.p2', 'K.m', 'K.ads',
+           'Q.max1', 'Q.max2', 'K.des', 'p.ep', 'p.em', 'r.ep',  'r.em')
+  missing <- setdiff(req, parms$parameter)
+  assert_that(length(missing) == 0, msg = paste('parms missing values for: ', paste0(missing, collapse = ', ')))
+
+
+  # Format the parameters into a vector.
+  p        <- parms$value
+  names(p) <- parms$parameter
+
+  with(as.list(c(state, p)), {
+
+    fxn_list <- list(
+      "F1d1" = function(){
+        # Microbial uptake from D1 pool
+        V.d1 * B * D1 /( K.d1 + B) / ( K.d1 + D1)
+      },
+      "F1d2" = function(){
+        # Microbial uptake from D2 pool
+        V.d2 * B * D2 /( K.d2 + B) /( K.d2 + D2)
+      },
+      "F2p1" = function(){
+        # Decomposition of P1 pool
+        V.p1 * EP1 * P1 / (K.p1 + P1)
+      },
+      "F2p2" = function(){
+        # Decomposition of P2 pool
+        V.p2 * EP2 * P2 / (K.p2 + P2)
+      },
+      "F3" = function(){
+        # Break down of mineralized organic carbon
+        V.m * EM * M / (K.m + M)
+      },
+      "F6d1" = function(){
+        # Adsorption of DOC 1 pool to mineral-associated organic carbon
+        K.ads * D1 *(1- Q1/ Q.max1)
+      },
+      "F6d2" = function(){
+        # Adsorption of DOC 2 pool to mineral-associated organic carbon
+        K.ads * D2 *(1- Q2/ Q.max2)
+      },
+      "F7d1" = function(){
+        # Desorption of mineral-associated organic carbon to DOC pool1
+        K.des * Q1/Q.max1
+      },
+      "F7d2" = function(){
+        # Desorption of mineral-associated organic carbon to DOC pool 2
+        K.des * Q2/Q.max2
+      },
+      "F8" = function(){
+        # Carbon loss due to microbial biomass mortality
+        (1 - p.ep - p.ep - p.em) * V.d1 * E.c1 * B * 0.5
```

Comment:
Why is this multiplied by 0.5 here in the millennial version of the model whereas in all MENDplus, MENDplus_ECA,  MENDp_reverse this flux is multiplied by 0.3?